### PR TITLE
Add Doxygen API docs

### PR DESCRIPTION
**Pull request by jmklix**
*Issue #, if available:*
https://github.com/aws/aws-iot-device-sdk-cpp-v2/issues/169

*Description of changes:*
Add api docs generated by doxygen

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,6 +76,7 @@ The best way to interact with our team is through GitHub. You can [open an issue
 Please make sure to check out our resources too before opening an issue:
```
[the remark this answers is not in the document]
Comment (jmklix) [reply README.md:77]: added mainpage.md for this

File: docs/html/____init_____8py.html [hunk 2]
```diff
@@ -0,0 +1,79 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "https://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
```
[the remark this answers is not in the document]
Comment (jmklix) [reply docs/html/____init_____8py.html]: I turned off the latex gen so it no longer exists

File: README.md [hunk 3]
```diff
@@ -76,6 +76,7 @@ The best way to interact with our team is through GitHub. You can [open an issue
...
 
 * Our [FAQ](FAQ.md)
```
[the remark this answers is not in the document]
Comment (jmklix) [reply README.md:78]: that didn't work, but I did filter them out with some manual exclusions

**Comment by TingDaoK** [comment]
It looks good! Thanks for fixing it!
Two things I think we can do better:
- Maybe add a script to generate it? We can add it to our release pipeline to update the docs automatically, like what we did for [aws-crt-cpp](https://github.com/awslabs/aws-crt-cpp/tree/main/docsrc)
- Maybe we should link the crt stuff back to the aws-crt-cpp docs, instead of generating it again? Like python, we have the sdk doc pointing back to aws-crt-python doc, which is nicer. But, the aws-crt-cpp doc seems broken now, and if it's too much work, we can put it into the backlog.

**Review by graebm — APPROVED** [review]

**Pull request merged**